### PR TITLE
[6.x] GitHub actions

### PR DESCRIPTION
**Pull request by driesvints**
For browser-kit-testing I'm sending this to master since we're about to release 6.x

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        php: [7.2, 7.3, 7.4]
+        laravel: [7.*]
+
+    name: P${{ matrix.php }} - L${{ matrix.laravel }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v1
+
+      - name: Cache dependencies
```
Comment (GrahamCampbell) [inline .github/workflows/tests.yml]: Does this actually make a difference. Does storing and fetching the cache take longer than just downloading the zips within composer install?

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        uses: actions/cache@v1
+        with:
+          path: ~/.composer/cache/files
+          key: dependencies-laravel-${{ matrix.laravel }}-php-${{ matrix.php }}-composer-${{ hashFiles('composer.json') }}
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v1
+        with:
+          php-version: ${{ matrix.php }}
+          extension-csv: dom, curl, libxml, mbstring, zip, pcntl, pdo, sqlite, pdo_sqlite, bcmath, soap, intl, gd, exif, iconv, imagick
```
Comment (GrahamCampbell) [inline .github/workflows/tests.yml]: We don't need a lot of these?

File: .github/workflows/tests.yml (hunk 1 above)
Comment (driesvints) [reply .github/workflows/tests.yml]: I took this over from Spatie's medialibrary so I guess this should work. Why else would it exist? :)

File: .github/workflows/tests.yml (hunk 2 above)
Comment (driesvints) [reply .github/workflows/tests.yml]: Took these over from Spatie's file. I don't want to figure out which ones we need or not so I'm just gonna consistently add these across the repos.

**Comment by GrahamCampbell** [comment]
image

**Comment by driesvints** [comment]
@GrahamCampbell fixed

**Comment by GrahamCampbell** [comment]
Hmm, adding all these extensions makes a major difference to the build time. Setting them up takes longer than running composer and the test suite?

**Comment by GrahamCampbell** [comment]
Also `shivammathur/setup-php@v1` is outdated. There is a v2 now?

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      fail-fast: true
+      matrix:
+        php: [7.2, 7.3, 7.4]
+        laravel: [^7.0]
+
+    name: P${{ matrix.php }} - L${{ matrix.laravel }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v1
```
Comment (GrahamCampbell) [inline .github/workflows/tests.yml]: Can this be upgraded to `v2`.

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+      - name: Cache dependencies
+        uses: actions/cache@v1
+        with:
+          path: ~/.composer/cache/files
+          key: dependencies-laravel-${{ matrix.laravel }}-php-${{ matrix.php }}-composer-${{ hashFiles('composer.json') }}
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v1
```
Comment (GrahamCampbell) [inline .github/workflows/tests.yml]: Same.

**Comment by driesvints** [comment]
> Hmm, adding all these extensions makes a major difference to the build time. Setting them up takes longer than running composer and the test suite?

Feel free to send in a follow up pr if you want.

File: .github/workflows/tests.yml (hunk 3 above)
Comment (driesvints) [reply .github/workflows/tests.yml]: Done

File: .github/workflows/tests.yml (hunk 4 above)
Comment (driesvints) [reply .github/workflows/tests.yml]: Done

**Comment by GrahamCampbell** [comment]
> Feel free to send in a follow up pr if you want.

Other than the laravel/framework repo, we can probably drop almost all the extensions.

**Comment by GrahamCampbell** [comment]
https://github.com/laravel/browser-kit-testing/pull/125 seems much faster on the "setup php" step.

**Comment by GrahamCampbell** [comment]
2x speed up on the whole run.

**Pull request merged**